src/components/contextual/pages/pool/PoolChart.vue

```diff
@@ -21,198 +23,309 @@ type Props = {
-  isStablePhantom(props.pool.poolType)
-);
+const currentChartValue = ref('');
+const currentChartDate = ref('');
+const isFocusedOnChart = ref(false);
 
+/**
+ * COMPUTED
+ */
 const appLoading = computed(() => store.state.app.loading);
 
-const history = computed(() => {
-  if (!props.historicalPrices) return [];
+const snapshotValues = computed(() => Object.values(props.snapshots || []));
 
-  const pricesTimestamps = Object.keys(props.historicalPrices);
-  const snapshotsTimestamps = Object.keys(props.snapshots);
+const periodOptions = computed(() => {
+  const maxPeriodLengh = snapshotValues.value.length;
+  const arr = [{ text: t('poolChart.period.all'), value: maxPeriodLengh }];
 
-  if (snapshotsTimestamps.length === 0) {
-    return [];
+  if (maxPeriodLengh > 365) {
+    arr.unshift({ text: t('poolChart.period.days', [365]), value: 365 });
   }
-
-  // Prices are required when not using pool liquidity
-  if (!supportsPoolLiquidity.value && pricesTimestamps.length === 0) {
-    return [];
+  if (maxPeriodLengh > 180) {
+    arr.unshift({ text: t('poolChart.period.days', [180]), value: 180 });
+  }
+  if (maxPeriodLengh > 90) {
+    arr.unshift({ text: t('poolChart.period.days', [90]), value: 90 });
   }
 
-  return snapshotsTimestamps
-    .map(snapshotTimestamp => {
-      const timestamp = parseInt(snapshotTimestamp);
-
-      const snapshot = props.snapshots[timestamp];
-      const prices = props.historicalPrices[timestamp] ?? [];
-      const amounts = snapshot.amounts ?? [];
-      const totalShares = parseFloat(snapshot.totalShares) ?? 0;
-      const liquidity = parseFloat(snapshot.liquidity) ?? 0;
-
-      return {
-        timestamp,
-        prices,
-        amounts,
-        totalShares,
-        liquidity
-      };
-    })
-    .filter(({ totalShares, prices, amounts, liquidity }) => {
-      if (!supportsPoolLiquidity.value && prices.length === 0) {
-        return false;
-      } else if (supportsPoolLiquidity.value && liquidity === 0) {
-        return false;
-      }
-      return totalShares > 0 && amounts.length > 0;
-    })
-    .reverse();
+  return arr;
 });
 
+const currentPeriod = ref(periodOptions.value[0].value || 90);
+
 const timestamps = computed(() =>
-  history.value.map(state => format(state.timestamp, 'yyyy/MM/dd'))
+  snapshotValues.value.map(snapshot => format(snapshot.timestamp, 'yyyy/MM/dd'))
 );
 
-const hodlValues = computed(() => {
-  if (history.value.length === 0) {
-    return [];
+const chartData = computed(() => {
+  const periodSnapshots =
+    currentPeriod.value === snapshotValues.value.length
+      ? snapshotValues.value
+      : snapshotValues.value.slice(0, currentPeriod.value - 1);
+  const isAllTimeSelected =
+    periodSnapshots.length === snapshotValues.value.length;
+  const pariodLastSnapshotIdx = periodSnapshots.length - 1;
+
+  if (activeTab.value === PoolChartTab.TVL) {
+    const tvlValues = periodSnapshots.map((snapshot, idx) => ({
```

Comment:
Each of these if statements could be put into a separate function.

Reply:
fixed